compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -4823,6 +4837,9 @@ private ParserRuleContext getNextRuleForIdentifier() {
             case TYPE_REFERENCE:
                 endContext();
                 return getNextRuleForTypeReference();
+            case SIMPLE_NAME_REFERENCE_IN_TYPE_DESC:
+                endContext();
+                return ParserRuleContext.TYPEDESC_RHS;
```

Comment:
I think we can remove starting a context of `SIMPLE_NAME_REFERENCE_IN_TYPE_DESC`. using `SIMPLE_NAME_REF_IDENTIFIER` alias will be sufficient

Reply:
`IDENTIFIER` has `MISSING IDENTIFIER` diagnostics right? We need to give `MISSING TYPE DESC`.
Problem with not having `SIMPLE_NAME_REFERENCE_IN_TYPE_DESC` ctx is; in `getNextRuleForIdentifier()` it may or may not go for the  `TYPEDESC_RHS`?

Reply:
I c we need to use alias `SIMPLE_NAME_REF_IDENTIFIER`. but still not clear about using `SIMPLE_NAME_REFERENCE_IN_TYPE_DESC`. what do u mean by may or may not go for the `TYPEDESC_RHS`?

Reply:
Addressed in d40178f :P